sbioapputils/app_runner/yaml_automation.py

```diff
@@ -173,19 +244,71 @@ def input_yaml_from_args(parameters):
     return input_settings
 
 
-def parameters_yaml_from_args(files: List[BytesIO], filenames: List[str]):
-    parameters = {}
-    library_found = False
-    
-    for file in files:
-        file_lines, argument_parsing_library = _parse_input_python(file)
-        if argument_parsing_library is not None:
-            library_found = True
-            new_parameters = _dict_from_args(file_lines, argument_parsing_library)
-            parameters = {**parameters, **new_parameters}
-    formatted_parameters = _format_argparse_parameters(parameters) if library_found else parameters
+def validate_multiple_outputs(outputs):
+    checkpoint = 0
+    not_yaml = 0
+    valid_outputs = []
+    for out in outputs:
+        valid = True
+        if any(substring in out for substring in ['directory','Directory','checkpoint','Checkpoint']):
+            checkpoint+=1
+            valid = False
+        if not is_invalid_yaml(out):
+            not_yaml+=1
+            valid = False
+        if valid:
+            valid_outputs.append(out)
+    return(valid_outputs)
+
 
+def parameters_yaml_from_args(files: List[BytesIO], filenames: List[str], method = 'new'):
+    
+    input_method = method
```

Comment:
It is confusing that the code saves method to a different variable and then changes the original variable. It is better to split the code for chatgpt parsing and manual parsing to different functions.
Then do the following logic:
```
if method is chatgpt parsing:
    call chatgpt parsing function
except:
    call manual parsing
else:
    call manual parsing
```
Same for the input_settings